native/src/main/java/io/ballerina/stdlib/sql/datasource/SQLDatasource.java

```diff
@@ -415,6 +420,22 @@ private String buildErrorMessage (Throwable t) {
         return message.toString();
     }
 
+    public void setExecuteGKFlag(boolean flag) {
+        this.executeGKFlag = flag;
+    }
+
+    public void setBatchExecuteGKFlag(boolean flag) {
+        this.batchExecuteGKFlag = flag;
+    }
```

Comment:
Shall we make these set methods private? allow only to change through `retrieveDatasource`